engines/sci/sound/audio32.cpp

```diff
@@ -0,0 +1,920 @@
+	if (channelIndex == kNoExistingChannel) {
+		return false;
+	}
+
+	Common::StackLock lock(_mutex);
+	const uint32 now = g_sci->getTickCount();
+
+	if (channelIndex == kAllChannels) {
+		// Global pause in SSCI is an extra layer over
+		// individual channel pauses, so only unpause channels
+		// if there was not a global pause in place
+		if (_pausedAtTick == 0) {
+			return false;
+		}
+
+		for (int i = 0; i < _numActiveChannels; ++i) {
+			AudioChannel &channel = getChannel(i);
+			if (!channel.pausedAtTick) {
+				channel.startedAtTick += now - _pausedAtTick;
+			}
+		}
+
+		_startedAtTick += now - _pausedAtTick;
+		_pausedAtTick = 0;
+		_mixer->pauseHandle(_handle, false);
+		return true;
+	} else if (channelIndex == kRobotChannel) {
+		for (int i = 0; i < getNumActiveChannels(); ++i) {
+			AudioChannel &channel = getChannel(i);
+			if (channel.robot) {
+				channel.startedAtTick += now - channel.pausedAtTick;
+				channel.pausedAtTick = 0;
+				// TODO: Robot
+				// StartRobot();
+				return true;
+			}
+		}
+	} else {
+		AudioChannel &channel = getChannel(channelIndex);
+		if (channel.pausedAtTick) {
+			channel.startedAtTick += now - channel.pausedAtTick;
+			channel.pausedAtTick = 0;
+			return true;
+		}
+	}
+
+	return false;
+}
+
+bool Audio32::pause(const int16 channelIndex) {
+	if (channelIndex == kNoExistingChannel) {
+		return false;
+	}
+
+	Common::StackLock lock(_mutex);
+	const uint32 now = g_sci->getTickCount();
+	bool didPause = false;
+
+	if (channelIndex == kAllChannels) {
+		if (_pausedAtTick == 0) {
+			_pausedAtTick = now;
+			_mixer->pauseHandle(_handle, true);
+			didPause = true;
+		}
+	} else if (channelIndex == kRobotChannel) {
+		_robotAudioPaused = true;
+		for (int16 i = 0; i < _numActiveChannels; ++i) {
+			AudioChannel &channel = getChannel(i);
+			if (channel.robot) {
+				channel.pausedAtTick = now;
+			}
+		}
+
+		// NOTE: The actual engine returns false here regardless of whether
+		// or not channels were paused
+	} else {
+		AudioChannel &channel = getChannel(channelIndex);
+
+		if (channel.pausedAtTick == 0) {
+			channel.pausedAtTick = now;
+			didPause = true;
+		}
+	}
+
+	return didPause;
+}
+
+int16 Audio32::stop(const int16 channelIndex) {
+	Common::StackLock lock(_mutex);
+	const int16 oldNumChannels = _numActiveChannels;
+
+	if (channelIndex == kNoExistingChannel || oldNumChannels == 0) {
+		return 0;
+	}
+
+	if (channelIndex == kAllChannels) {
+		for (int i = 0; i < oldNumChannels; ++i) {
+			freeChannel(i);
+		}
+		_numActiveChannels = 0;
+	} else {
+		freeChannel(channelIndex);
+		--_numActiveChannels;
+		for (int i = channelIndex; i < oldNumChannels - 1; ++i) {
+			_channels[i] = _channels[i + 1];
+			if (i + 1 == _monitoredChannelIndex) {
+				_monitoredChannelIndex = i;
+			}
+		}
+	}
+
+	// NOTE: SSCI stops the DSP interrupt and frees the
+	// global decompression buffer here if there are no
+	// more active channels
+	if (_numActiveChannels == 0) {
+		_mixer->pauseHandle(_handle, true);
```

Comment:
This indicates that the audio should be stopped.... so why pause the handle, instead of stopping it?